src/Tools/Program.cs

```diff
@@ -0,0 +1,337 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml;
+using Microsoft.Build.Locator;
+using Microsoft.CodeAnalysis.CodeMetrics;
+using Microsoft.CodeAnalysis.MSBuild;
+
+namespace Metrics
+{
+    internal sealed class Program
+    {
+        public static int Main(string[] args)
+        {
+            var task = RunAsync(args);
+            task.Wait();
+            return (int)task.Result;
```

Comment:
any reason using this pattern? does this have some magic behavior than .Result or GetAwaiter().GetResult()?

just curious.

by the way, if user do ctrl+c in command line, will this console app get break out? otherwise, a way to connect ctrl+c to cancecllation and pass in cancellation to Wait or other?

Reply:
No specific reason. I will see if I can thread in cancellation token.